harry-core/src/harry/reconciler/Reconciler.java

```diff
@@ -87,31 +87,36 @@ class Processor extends VisitExecutor
         {
             // Whether a partition deletion was encountered on this LTS.
             private boolean hadPartitionDeletion = false;
+            private boolean hadTrackingRowWrite = false;
             private final List<Ranges.Range> rangeDeletes = new ArrayList<>();
-            private final List<ReplayingVisitor.Operation> writes = new ArrayList<>();
-            private final List<ReplayingVisitor.Operation> columnDeletes = new ArrayList<>();
+            private final List<Operation> writes = new ArrayList<>();
+            private final List<Operation> columnDeletes = new ArrayList<>();
 
             @Override
-            protected void operation(long lts, long pd, long cd, long opId, OpSelectors.OperationKind opType)
+            protected void operation(Operation operation)
             {
                 if (hadPartitionDeletion)
                     return;
 
-                switch (opType)
+                long lts = operation.lts();
+                assert pdSelector.pd(operation.lts(), schema) == operation.pd();
+
+                if (operation.opKind() != OpSelectors.OperationKind.DELETE_PARTITION)
+                    partitionState.visitedLts.add(operation.lts());
```

Comment:
Deleting the whole partition doesn't count as a visit?

Reply:
extracted into a method and documented:

```

        /**
         * All operations apart from partiton delition, including partition hist are visible since if there is a
         * partition liveness marker, partition's static column is going to survive. We use this method to match
         * computed LTS with tracked ones.
         */
        public boolean hasVisibleVisit()
        {
            return this != OpSelectors.OperationKind.DELETE_PARTITION;
        }
```